assets/js/src/integrations/classic.js

```diff
@@ -156,4 +158,67 @@ export function trackClassicPages( {
 				} );
 			} );
 		} );
+
+	// Handle selection and add_to_cart in **Block** product listing the classic way.
+	// Attach click event listeners to a whole product card, as some links may not have the product_id data attribute.
+	document
+		.querySelectorAll( '.products-block-post-template .product' )
+		?.forEach( ( productCard ) => {
+			// Get the Product ID from a child node containing the relevant attribute
+			const productId = productCard
+				.querySelector( '[data-product_id]' )
+				?.getAttribute( 'data-product_id' );
+
+			if ( ! productId ) {
+				return;
+			}
+
+			productCard.addEventListener( 'click', ( event ) => {
+				const target = event.target;
+				// `product-view-link` has no serilized HTML identifier/selector, so we look for the parent block element.
+				const viewLink = target.closest(
+					'.wc-block-components-product-image a'
+				);
+
+				// Catch name click
+				const nameLink = target.closest( '.wp-block-post-title a' );
+
+				// Catch the enclosing product button.
+				const button = target.closest(
+					'.wc-block-components-product-button [data-product_id]'
+				);
+
+				const isAddToCartButton =
+					button &&
+					button.classList.contains( 'add_to_cart_button' ) &&
+					! button.classList.contains( 'product_type_variable' );
+
+				if ( isAddToCartButton ) {
+					// Add to cart.
+					tracker.eventHandler( 'add_to_cart' )( {
+						product: getProductFromID(
+							parseInt( productId ),
+							products,
+							cart
+						),
+					} );
+				} else if ( viewLink || button || nameLink ) {
+					// Product image or add-to-cart-like button.
+					tracker.eventHandler( 'select_content' )( {
+						product: getProductFromID(
+							parseInt( productId ),
+							products,
+							cart
+						),
+					} );
+				}
+			} );
+		} );
+
+	// Track View Item List
+	if ( products?.length ) {
+		tracker.eventHandler( 'view_item_list' )( {
+			products,
+		} );
+	}
```

Comment:
:grey_question: 
The biggest concern I have is regarding this bit.
Other "on-page-load" events are handled through `events` object. I'm concerned not only about consistency, but also about firing an event twice.

I'd try reproducing and test how to achieve that live, but it would be nice to have E2Es in place https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/385

Reply:
The E2E test handle checking if the event is sent. But since it waits for all requests and then returns the one that matches. I haven't found a way yet to confirm it doesn't send an event twice.

Reply:
I https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/378/files#diff-ca051807ffd7c39f888ccb191f645da9c2f795acc8f63eb4b0a87f6da5da0bbf `removeAction( `${ ACTION_PREFIX }-view_item_list`, NAMESPACE );` as well as https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/378/files#diff-feb1d32ea04490583d36ba93f2101db973aad5303bbabee4037af3767ac3711e `'view_item_list' => 'woocommerce_before_shop_loop_item',` from the mapping.

So I guess it won't trigger anywhere else that when product data is filled.

Reply:
I mean that we already have a logic to track events on page load. It's 
https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/c34d9325a9eca2aa7b93b78c3f8c1bd9d6e29594/assets/js/src/integrations/classic.js#L32:L45

Adding those few lines adds yet another place with almost the same responsibility, which would lead to future duplications, as we may miss one or the other.

I think we could avoid adding this bit in JS, and simply bring back `woocommerce_loop_add_to_cart_link` to the mappings list, or add it manually on the `woocommerce_loop_add_to_cart_link` callback. If we're adding a product to `products`, we could as well at the same time add `view_item_list` to `events`.

This way we will have one predicate to determine what to call on the pageload: "an event from `events` array".

Reply:
Like in https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/378/commits/6ad4979e152f9986e551889042e27bd7d435ef14